pkg/applier/kpt_applier.go

```diff
@@ -512,57 +514,60 @@ func (a *Applier) sync(ctx context.Context, objs []client.Object) (map[schema.Gr
 		}
 		gvks[resource.GetObjectKind().GroupVersionKind()] = struct{}{}
 	}
-	if a.errs == nil {
+
+	errs := a.Errors()
+	if errs == nil {
 		klog.V(4).Infof("all resources are up to date.")
 	}
-
 	if s.Empty() {
 		klog.V(4).Infof("The applier made no new progress")
 	} else {
 		klog.Infof("The applier made new progress: %s", s.String())
 		objStatusMap.Log(klog.V(0))
 	}
-	return gvks, a.errs
+	return gvks, errs
 }
 
+// Errors returns the errors encountered during the last apply or current apply
+// if still running.
 // Errors implements Interface.
-// Errors returns the errors encountered during apply.
 func (a *Applier) Errors() status.MultiError {
-	// TODO: Make read/write of a.errs thread-safe
-	// Return a copy
+	a.errorMux.RLock()
+	defer a.errorMux.RUnlock()
+
+	// Return a copy to avoid persisting caller modifications
 	return status.Append(nil, a.errs)
```

Comment:
why not use `return a.errs`?

Reply:
to avoid persisting modifications by any caller that might append additional errors to the MultiError.